helm/minio/templates/servicemonitor.yaml

```diff
@@ -28,9 +28,6 @@ spec:
       {{- if .Values.metrics.serviceMonitor.scrapeTimeout }}
       scrapeTimeout: {{ .Values.metrics.serviceMonitor.scrapeTimeout }}
       {{- end }}
-      bearerTokenSecret:
-        name: {{ template "minio.fullname" . }}-prometheus
-        key: token
```

Comment:
why is this a problem? @dyvol777 since all services point to same deployment?

Reply:
ServiceMonitor create targets for prometheus.
If there are bearer token, you need manual actions (generate token and create secret). 
With `MINIO_PROMETHEUS_AUTH_TYPE: public` there is no need to use bearer token, so no manual actions is required

Reply:
How about adding this as a feature for `serviceMonitor` i.e `serviceMonitr.public= true`

Reply:
Reworked in new commit

Reply:
@harshavardhana